lib/Dialect/EmitC/Conversion/TosaToEmitC.cpp

```diff
@@ -233,6 +233,35 @@ class MatMulOpConversion : public OpConversionPattern<tosa::MatMulOp> {
   }
 };
 
+/// Convert `tosa.negate` into an `emitc.call` operation.
+class NegateOpConversion : public OpConversionPattern<tosa::NegateOp> {
+  using OpConversionPattern<tosa::NegateOp>::OpConversionPattern;
+
+public:
+  NegateOpConversion(MLIRContext *ctx)
+      : OpConversionPattern<tosa::NegateOp>(ctx) {}
+
+private:
+  LogicalResult
+  matchAndRewrite(tosa::NegateOp negateOp, ArrayRef<Value> operands,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (negateOp.quantization_info().hasValue()) {
```

Comment:
Is this the cause for the custom OpConversion?

Reply:
Yes, it's solely to forbid the quantization.